tests/integration/gateway/main_test.go

```diff
@@ -0,0 +1,148 @@
+// +build integ
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package gateway
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+const (
+	ASvc                 = "a" // Used by the default gateway
+	BSvc                 = "b" // Used by the custom ingress gateway
+	CustomServiceGateway = "custom-ingressgateway"
+)
+
+type EchoDeployments struct {
+	appANamespace, appBNamespace namespace.Instance
+	A, B                         echo.Instances
+}
+
+var (
+	Inst              istio.Instance
+	CgwInst           istio.Instance
+	Apps              = &EchoDeployments{}
+	CustomGWNamespace namespace.Instance
+)
+
+func TestMain(m *testing.M) {
```

Comment:
should we just deploy the gateway in another test as part of the main suite? There shouldn't be need for custom test suite (expesnive)

Reply:
I believe I can move this somewhere else. For now, having it by itself allows me to verify that things work by running a single test directory.

Reply:
Moved under pilot suite.